File: Year II/First Semester/MAP/Laborator/Laborator2/factory/TaskContainerFactory.java
package sem1_2.factory;

import sem1_2.model.Container;
import sem1_2.model.StackContainer;
import sem1_2.model.QueueContainer;

public class TaskContainerFactory implements Factory {
    private static TaskContainerFactory factory;

    private TaskContainerFactory(){}

    @Override
    public Container createContainer(Strategy strategy) {
        if (strategy == Strategy.FIFO) {
            return new QueueContainer();
        } else {
            return new StackContainer();
        }
    }

    // singleton
    public static TaskContainerFactory getInstance(){
        if(factory == null){
            factory = new TaskContainerFactory();
        }
        return factory;
    }
}

File: Year II/First Semester/MAP/Laborator/Laborator9/ConsoleApp1/ConsoleApp1/Factory/TaskContainerFactory.cs
using ConsoleApp1.Model;

namespace ConsoleApp1.Factory;

public class TaskContainerFactory: Factory
{
    private TaskContainerFactory() { }
    private static TaskContainerFactory _instance;
    public static TaskContainerFactory FactoryInstance => _instance ??= new TaskContainerFactory();

    public Container createContainer(Strategy strategy)
    {
        if (strategy == Strategy.Lifo)
            return new StackContainer();
        return new QueueContainer();
    }
    
}